### DISCUSSION TITLE
Accessing DSC Zone Bypass Switch

### DISCUSSION THREAD

**Discussion by bluepixel00** [ufodone/envisalink_new#136] [Q&A] [answered]
I have looked around and not seeing how to enable or access the zone bypass switches.  Has anybody been able to access these on a dsc panel.

Ufodone, By the way thanks for all the work on project.  I have been using the original add-on for years and recently moved to the “new” addon.

**Comment by ufodone** [comment, marked as answer]
You can enable the zone bypass switch via `Settings->Devices & Services -> envisalink_new -> Configure -> Advanced`.  This is turned off by default because it only works if your panel is setup to NOT require a passcode to bypass a zone.  Once you've enabed the "Create zone bypass switches" option, it will create a new switch entity for each zone.

**Reply by bluepixel00** [reply]
Thanks!  I was staring at that toggle and in my head tied that the wireless zones!  Thanks again

**Reply by bluepixel00** [reply]
Thanks again for all the work on this integration!

**Reply by elechtma** [reply]
Hi,
I've tried to do this, but I don't see that option under advanced all I see are 3 fields:
- panic type
- keep alive interval
- connection timeout

... update: reinstalling the integration seems to have fixed it!

**Reply by lsibarra** [reply]
Hi,

I’m new to Home Assistant and having issues with the envisalink_new integration.
I deleted it via Settings → Devices & Services → envisalink_new → Configure → Delete, then tried adding it again by entering my info, but it always fails.

The first time, it migrated my old envisalink setup from configuration.yaml, but the zone bypass switch was missing. I read a post saying reinstalling the integration would fix it, but now it won’t set up at all and the switch is still missing.

Do I need to completely remove the add-on from HACS, re-download, and reinstall it? Is that what’s meant by “reinstalling the integration”?

Thanks,

**Reply by ufodone** [reply]
The zone bypass switches are turned off by default and you need to explicitly enable them in the Advanced configuration as per above.  But there is a bug where the panel type doesn't get saved properly so the configuration doesn't think it's a DSC panel and so the option to enable the bypass switch doesn't show up.  I've been unable to reproduce this myself though so haven't been able to fix the issue yet.

> but now it won’t set up at all and the switch is still missing.

Can you elaborate on this?  Are you trying to have it re-import from configuraiton.yaml or were you trying to explicitly add it via the UI?  What errors, etc. are you seeing when you try and set it up?  Is there anything in the logs?

**Reply by lsibarra** [reply]
Hi @ufodone 

Thank you for your reply. Do you think the issue might be caused by the old Envisalink configuration still being in my YAML file? I left that configuration in place, then downloaded this repository as a custom integration and installed it. After restarting Home Assistant, I went to Add Integration, entered the basic information, and submitted it, but received this error:

<img width="1036" height="344" alt="image" src="https://github.com/user-attachments/assets/5bdb0c03-1208-4465-9931-729fed739698" />

After reinstalling the integration, the error would not clear and no entities were created. Today, I decided to delete both the integration and the envisalink_new add-on from HACS, then start over. Similar to my earlier attempt, the first setup produced the same error, but after a reboot, it corrected itself and created the entities. However, I’m still missing the Zone Bypass Switch option under Settings → Advanced → Options.

<img width="413" height="496" alt="image" src="https://github.com/user-attachments/assets/e9692894-35e9-42da-9f19-72b077df5881" />

I am willing to help identify the bug. Please let me know what I need to do to help with this. I just enabled the debug logging for envisalink_new and rebooted the system. After the reboot, the devices/entities failed to load

<img width="1043" height="478" alt="image" src="https://github.com/user-attachments/assets/a600237c-47d9-4be8-8d02-4319819acdad" />

Here are the logs I was able to find.

`Logger: custom_components.envisalink_new.pyenvisalink.envisalink_base_client
Source: custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py:138
integration: EyezOn ([documentation](https://github.com/ufodone/envisalink_new), [issues](https://github.com/ufodone/envisalink_new/issues))
First occurred: 8:39:32 PM (5 occurrences)
Last logged: 8:41:04 PM

Caught unexpected exception: ConnectionResetError(104, 'Connection reset by peer')`

`Logger: custom_components.envisalink_new.pyenvisalink.envisalink_base_client
Source: custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py:210
integration: EyezOn ([documentation](https://github.com/ufodone/envisalink_new), [issues](https://github.com/ufodone/envisalink_new/issues))
First occurred: 8:39:32 PM (5 occurrences)
Last logged: 8:41:04 PM

Exception while closing connection: [Errno 104] Connection reset by peer`

`Logger: custom_components.envisalink_new.pyenvisalink.envisalink_base_client
Source: custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py:144
integration: EyezOn ([documentation](https://github.com/ufodone/envisalink_new), [issues](https://github.com/ufodone/envisalink_new/issues))
First occurred: 8:39:32 PM (5 occurrences)
Last logged: 8:41:04 PM

Reconnection attempt in 30s`

Hope this helps.

**Reply by ufodone** [reply]
You definitely need to remove or comment out the `envisalink` section from your configuration.yaml and then restart Home Assistant for this to work.  The EVL devices only allow a single connection to them at a time so if you have your `envisalink` integration enabled (by having the config in place), it is likely taking that one connection and causing those errors.  That `Connection reset by peer` error is a good indication that is probably what's happening.

What I'd recommend you try is:
1. Delete your device from the EyezOn integration.
2. Comment out the `envisalink` section from your configuration.yaml.
3. Restart Home Assistant. (At this point nothing should be connecting to the EVL device.)
4. Add your device back via the EyezOn integration.

That should hopefully resolve the conflict between the stock integration and this one and allow it to connection properly.

I'll also be curious to know if that resolves the zone bypass switch option not showing up.

**Comment by lsibarra** [comment]
Thank you so much, @ufodone, for your guidance. Following your advice, I commented out my Envisalink configuration in configuration.yaml, rebooted the system, and re-added the EyezOn integration. After entering my alarm system’s basic details, I received a pop-up confirming that it detected my EyezOn card (version 3) and the alarm name.

Once I clicked Finish, all entities appeared as expected. I then checked Settings → Advanced → Options and confirmed that the Create Zone Bypass Switches toggle is now available. See screenshot below.

<img width="394" height="721" alt="image" src="https://github.com/user-attachments/assets/03f9eeaa-820a-44ec-9be2-27f522ee69ee" />

Also, may I know if there’s a way to use the alarm_keypress method with this integration? I’d like to understand if it can be triggered directly from within Home Assistant, and if so, how to properly set it up.

Thanks in advance.

**Reply by lsibarra** [reply]
I tried using this in the automation but I see no options to select the Target to enter the custom bypass

<img width="1054" height="367" alt="image" src="https://github.com/user-attachments/assets/1cf0fee0-14b8-4780-9269-9935adf6b3cf" />

**Reply by ufodone** [reply]
Yes, `alarm_keypress` action is available with this integration.  If I recall correctly, that action moved from the integration level to the alarm control panel entity itself so you if you are trying to modify an existing automation that referenced the `envisalink` one, that may be why no target is showing up.  The yaml for the action would look something like this:

```
action: envisalink_new.alarm_keypress
target:
  entity_id: alarm_control_panel.home_alarm
data:
  keypress: "*1"
```

**Reply by ufodone** [reply]
Oh, and if you're just trying to bypass a zone, then you can just use `switch` action to change the appropriate zone bypass switch.  You don't need to use (and shouldn't) use a keypress action for that.

**Reply by lsibarra** [reply]
Thank you again, @ufodone. I was able to locate the alarm_keypress function. I grouped the bypass switches for the upstairs zones, but this process is taking longer and seems to trigger each zone individually. Since I’ve already grouped the upstairs zones in my alarm panel, I can use the alarm_keypress command to bypass the entire group at once. That’s why I asked—I want to include this group bypass in an automation.

**Reply by jasonbwilliams** [reply]
> Oh, and if you're just trying to bypass a zone, then you can just use `switch` action to change the appropriate zone bypass switch. You don't need to use (and shouldn't) use a keypress action for that.

Hi @ufodone.  Just converting over from the legacy Envisalink integration and I'm so thankful you built this!  I'm having trouble understanding how to use the bypass function.  I have a DSC1832 and the option in the advanced config is enabled.  I see all the switches, but when I turn on the zones I want to bypass, it won't let me arm.  I get the error "system is either not-secure, in exit-delay, or already armed) (024)".  Is this because those zones are already tripped?  Can you arm the system with windows already open using the bypass switch, or am I completely off base?

Thanks again for the work!

**Reply by ufodone** [reply]
> Can you arm the system with windows already open using the bypass switch

Yes, you should be able to.  Once you bypass a zone, that zone being faulted will not prevent the alarm from being armed and will not trigger the alarm if faulted after the system is armed.

The error you are getting indicates that the alarm is not in a "Ready" state for arming.  This typically means that there is a zone faulted - and in your case, likely that the bypass failed.

When you toggle the a zone's bypass switch, does it stay turned on?  Or does it switch back to the "off" position after a few seconds?  If the latter, then the panel is likely rejecting the command to toggle the zone's bypass state.  One thing that can cause this is if the physical panel itself is configured to require the alarm code to bypass a zone.  This is configured in Section [015] Option [5] (at least on my DSC system).  When you try and bypass a zone via a physical keypad (the `*1` command), are you required to enter the alarm code before toggling a zone for bypass?

**Reply by jasonbwilliams** [reply]
Got it working!  I discovered that I couldn't arm even without bypass, even directly on the Envisalink URL.  I reset the panel and voila!  Your integration should really replace the default in HA.  It's so easy to use!

Thanks again for the help.

**Comment by Vito7777** [comment]
hi, thank you so much for sharing your work. I'm new with envisalink.  I was looking attributes in zone entity and noticed that at least "bypassed" attribute doesn't refresh (I bypass zone using physical panel). The same with "tamper" attribute ( I have one zone  with tamper on)  What can be done in order to refresh statuses? 

Also bypass switch toggle back and doesn't really bypass.  error log: 
This error originated from a custom integration.
Logger: custom_components.envisalink_new.pyenvisalink.dsc_client
Source: custom_components/envisalink_new/pyenvisalink/dsc_client.py:297
integration: EyezOn (documentation, issues)
First occurred: 19:15:21 (2 occurrences)
Last logged: 19:15:31

The envisalink asked for a code, but we have no code in our cache.  

Entity alarm_control_panel.home_partition_1 have code

**Reply by Vito7777** [reply]
OK, perhaps this is because  DSC requires alarm code to bypass and  that could be done by scripting. Do you have function to emulate keypad press ?

So key question is why zone statuses do not refresh, why this error appears:

This error originated from a custom integration.
Logger: custom_components.envisalink_new.pyenvisalink.envisalink_base_client
Source: custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py:604
integration: EyezOn (documentation, issues)
First occurred: 19:16:42 (95 occurrences)
Last logged: 19:17:26

Command '071 1*122#' failed; retry in 0.4 seconds.
Command '071 1*122#' failed; retry in 0.8 seconds.
Command '071 1*109#' failed; retry in 0.2 seconds.
Command '071 1*109#' failed; retry in 0.4 seconds.
Command '071 1*109#' failed; retry in 0.8 seconds.

**Reply by Vito7777** [reply]
zones opening activity flow works ok

**Reply by ufodone** [reply]
The zone bypass switches currently only work on DSC panels that are setup to not require a code to bypass a zone.  I suspect that you aren't getting the status updates on zones bypassed from the physical panel either because the integration queries that using the `*1#` keystrokes which I believe will also fail if the panel is configured to require a code for bypass.

At the moment, the only way to get it working would be for you to change the configuration on your panel to disable needing a code for bypass - but I recognize that might not be something you want to do for your setup.

I think it may be possible to add support for systems requiring a code for bypass but I haven't been able to find the time to try it out yet.

**Reply by ufodone** [reply]
> The same with "tamper" attribute ( I have one zone with tamper on) 

It's possible this is related to the bypass issue.  If your system requires a code for bypass, you should disable the zone bypass switches as it a leave your physical panel in a weird state waiting for a code to be provided.

If you turn off the bypass switches and the tamper still isn't working, then please open an issue to track the problem.  Turn on debug logging for the integration and restart HA and then attach the log to the issue so I can have a look at the updates the EVL is sending.
